kamon-core/src/main/java/kamon/jsr166/Striped64.java

```diff
@@ -182,6 +186,13 @@ final boolean casBase(long cmp, long val) {
     }
 
     /**
+     * CASes the base field.
```

Comment:
If getAndSetLong was added too recently I'd suggest adding the function as a static helper, as it's needed for both the base get and the cell get.

The code you wrote is identical to the [actual impl](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/sun/misc/Unsafe.java#l1085).